test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -2019,6 +2019,75 @@ public int Serialize(ref byte[] bytes, int offset, T value, IFormatterResolver f
             }
         }
 
+        [Fact]
+        public async Task ConnectionNotTimedOutIfClientNeverPings()
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(services =>
+                services.Configure<HubOptions>(options =>
+                   options.ClientTimeoutInterval = TimeSpan.FromMilliseconds(100)));
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient(new JsonHubProtocol()))
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+                await client.Connected.OrTimeout();
+                // This is a fake client -- it doesn't auto-ping to signal
+
+                // We go over the 100 ms timeout interval...
+                await Task.Delay(120);
+                client.TickHeartbeat();
+
+                // but client should still be open, since it never pinged to activate the timeout checking
+                await Assert.ThrowsAsync<TimeoutException>(() => connectionHandlerTask.OrTimeout(60));
```

Comment:
Using a timeout to test that the client is still open is weird. Is there no flag on TestClient that says the state of the connection?

Reply:
Yeah, you can just check that the task has not completed here. The `Task.Delay(120)` followed by manually ticking the heartbeat means we know at least `120` milliseconds have elapsed (if not more)

```
Assert.False(connectionHandlerTask.IsCompleted)
```